packages/medusa/src/types/global.ts

```diff
@@ -33,12 +33,36 @@ export type Logger = _Logger & {
   warn: (msg: string) => void
 }
 
+export type DatabaseTlsOptions = {
+  ca: string | undefined
+  rejectUnauthorized: boolean | undefined
+}
+
+export type DatabaseHostConfig =
+  | {
+      database?: string
+      password?: string | (() => string) | (() => Promise<string>)
+      port?: number
+      host?: string
+      ssl?: boolean | DatabaseTlsOptions
+      username?: string
+    }
+  | {
+      database?: string
+      url?: string
+    }
+
 export type ConfigModule = {
   projectConfig: {
     redis_url?: string
 
     jwt_secret?: string
     cookie_secret?: string
+    database_host?: string
```

Comment:
Suggestion: i would suggest to do something like DatabaseHostConfig & { ... } for the projectConfig type

Reply:
I'm not certain about this change, coz it may need us to change medusa-config.js, which lives in a completely different repo, and could break some existing deployments.. wdyt ?

Reply:
Don't think it could break existing projects because previously you was only able to specify database and url, the type allows that and only raise an error if you provide along the previous one the other one